p2p/protocol/holepunch/coordination_test.go

```diff
@@ -103,7 +103,7 @@ func TestEndToEndSimConnect(t *testing.T) {
 	// ensure no hole-punching streams are open on either side
 	ensureNoHolePunchingStream(t, h1, h2)
 	events := tr.getEvents()
-	require.Len(t, events, 3)
+	require.Eventually(t, func() bool { return len(events) == 3 }, time.Second, 10*time.Millisecond)
```

Comment:
We need to repeately call `getEvents`, the returned slice is a copy.

Reply:
🤦🏻‍♂️ of course, thanks for catching this